spec/features/courses/results_spec.rb

```diff
@@ -0,0 +1,213 @@
+#https://github.com/JsonApiClient/json_api_client/tree/db890adf3d7175e91829d359336ab7c98ed90a34#paginating
+#https://jsonapi.org/format/#fetching-pagination
+#Information needed:
+# Provider name
+# Provider code
+# Course code
+# Course type (e.g: PGCE)
+# Financial support
+# Accredited body if applicable
+# Main address
+
+require "rails_helper"
+
+feature "Search results", type: :feature do
+  def jsonapi_site_status(name, study_mode, status)
+    build(:site_status, study_mode, site: build(:site, location_name: name), status: status)
+  end
+
+  let(:provider) do
+    build(:provider,
+          provider_name: "ACME SCITT A0",
+          provider_code: "T92",
+          website: "https://scitt.org",
+          address1: "1 Long Rd",
+          postcode: "E1 ABC")
+  end
+
+  let(:provider2) do
+    build(:provider,
+          provider_name: "ACME SCITT A0",
+          provider_code: "T92",
+          website: "https://scitt.org",
+          address1: "Building 64",
+          address2: "32 Copton Lane",
+          address3: "Bracknel",
+          address4: "Berkshire",
+          postcode: "NXT STP")
+  end
+
+  let(:course) do
+    build(:course,
+          name: "Primary",
+          course_code: "X130",
+          provider: provider,
+          provider_code: provider.provider_code,
+          recruitment_cycle: current_recruitment_cycle)
+  end
+
+  let(:current_recruitment_cycle) { build :recruitment_cycle }
+
+  let(:results_page) { PageObjects::Page::Results.new }
+
+  let(:subject) do
+    build(:subject,
+          :english,
+          scholarship: "2000",
+          bursary_amount: "4000",
+          early_career_payments: "1000")
+  end
+
+  let(:accrediting_provider) { build(:provider) }
+  let(:decorated_course) { course.decorate }
+  let(:courses) { [course] }
+  let(:courses_request) do
+    stub_api_v3_request(
+      "/recruitment_cycles/#{Settings.current_cycle}/providers/courses?include=provider,accrediting_provider&page%5Bpage%5D=1&page%5Bper_page%5D=10",
+      resource_list_to_jsonapi(courses, include: %i[provider accrediting_provider]),
+    )
```

Comment:
How about a `stub_api_v3_resource` based off of [`stub_api_v2_resource`](https://github.com/DFE-Digital/manage-courses-frontend/blob/de7378a5adc3d442db322c8aea6ceb267c37f7b9/spec/support/helpers.rb#L96-L114)?

Reply:
I think the `stub_api_v2_resource()` series of methods was at the wrong level of abstraction and complicated certain things by making assumptions such as that:  
 - there would always be a resource to stub  
 - `include` was the only argument we'd use
 - we'd only ever be getting data  
 - we'd never need to validate the data being sent  

These have lead to the method being complicated and doing too much, which meant alot of time was spent trying to re-engineer it to meet our new needs. As such I think we should wait until we have a better understanding of how we've used this API before we try to make another helper method like it.

Reply:
A lot of that is by design, tbh. `stub_api_v2_resource` is meant to be opinionated, for the sake of readability and convenience. It's really just a bit of sugar around specific types of requests: resources that can be retrieved. It's not meant to be a multi-tool that can handle all requests to the API.